traits/ctraits.c

```diff
@@ -2908,22 +2897,31 @@ static trait_setattr setattr_handlers[] = {
     /*  End of __setstate__ method entries */
     NULL};
 
-static int
-trait_init(trait_object *trait, PyObject *args, PyObject *kwds)
+
+trait_object *
+trait_new (PyTypeObject *trait_type, PyObject *args, PyObject *kw)
 {
-    int kind;
+    int kind, n;
+    trait_object *trait;
 
-    if (!PyArg_ParseTuple(args, "i", &kind)) {
-        return -1;
+    n = PyTuple_GET_SIZE(args);
```

Comment:
There's an easier way to handle this sort of optional argument: if you use `"|i"` instead of `"i"` as the format string in `PyArg_ParseTuple`, and initialise `kind` to `0` in the declaration, you can get rid of `n` and the extra branch.

Reply:
Thanks. I changed the code to match the suggestion.